src/util/prolog/prologResult.ts

```diff
@@ -9,22 +9,32 @@ const booleanAnswerRegExp = /^(true|false)( ;|,|.)$/;
 // Second group contains the variable names
 const variableNameRegExp = /(^| )([A-Z]\w*)/g;
 
+/** Handles the results for a prolog query returned by PrologSession.executeQuery() */
 export class PrologResult {
 	private readonly rawResults: string[];
 	private results: Map<string, Array<string | boolean>> | undefined;
 
+	/**
+		* Simply forwards the raw results to its respective attribute
+		* @param {Array<string>} rawResults - The string array containing the results
```

Comment:
To make documentation most useful, you don't need to describe the type or anything already contained in the name, but give additional information - e.g. what format those raw results should have, or where you would get them from (e.g. "The values received from ..." or "The prolog results in format ..."). All the documentation in this block I could have written without knowing what any of this code does; but I still don't know how to use this class / what to pass into the constructor.